proposals/bip-003.md

```diff
@@ -0,0 +1,460 @@
+
+The following roles are proposed for removal:
+
+* `SimpleIdentifier`
+* `QualifiedIdentifier`
+* `BinaryExpression`
+* `BinaryExpressionLeft`
+* `BinaryExpressionRight`
+* `BinaryExpressionOp`
+* `OpBitwiseLeftShift`
+* `OpBitwiseRightShift`
+* `OpBitwiseUnsignedRightShift`
+* `OpBitwiseOr`
+* `OpBitwiseXor`
+* `OpBitwiseAnd`
+* `OpEqual`
+* `OpNotEqual`
+* `OpLessThan`
+* `OpLessThanEqual`
+* `OpGreaterThan`
+* `OpGreaterThanEqual`
+* `OpSame`
+* `OpNotSame`
+* `OpContains`
+* `OpNotContains`
+* `OpPreIncrement`
+* `OpPostIncrement`
+* `OpPreDecrement`
+* `OpPostDecrement`
+* `OpNegative`
+* `OpPositive`
+* `OpBitwiseComplement`
+* `OpDereference`
+* `OpTakeAddress`
+* `OpBooleanAnd`
+* `OpBooleanOr`
+* `OpBooleanNot`
+* `OpBooleanXor`
+* `OpAdd`
+* `OpSubstract`
+* `OpMultiply`
+* `OpDivide`
+* `OpMod`
+* `PackageDeclaration`
+* `ImportDeclaration`
+* `ImportPath`
+* `ImportAlias`
+* `FunctionDeclaration`
+* `FunctionDeclarationBody`
+* `FunctionDeclarationName`
+* `FunctionDeclarationReceiver`
+* `FunctionDeclarationArgument`
+* `FunctionDeclarationArgumentName`
+* `FunctionDeclarationArgumentDefaultValue`
+* `FunctionDeclarationVarArgsList`
+* `TypeDeclaration`
+* `TypeDeclarationBody`
+* `TypeDeclarationBases`
+* `TypeDeclarationImplements`
+* `VisibleFromInstance`
+* `VisibleFromType`
+* `VisibleFromSubtype`
+* `VisibleFromPackage`
+* `VisibleFromSubpackage`
+* `VisibleFromModule`
+* `VisibleFromFriend`
+* `VisibleFromWorld`
+* `IfCondition`
+* `IfBody`
+* `IfElse`
+* `SwitchCase`
+* `SwitchCaseCondition`
+* `SwitchCaseBody`
+* `SwitchDefault`
+* `ForInit`
+* `ForExpression`
+* `ForUpdate`
+* `ForBody`
+* `ForEach`
+* `WhileCondition`
+* `WhileBody`
+* `DoWhileCondition`
+* `DoWhileBody`
+* `TryBody`
+* `TryCatch`
+* `TryFinally`
+* `CallReceiver`
+* `CallCallee`
+* `CallPositionalArgument`
+* `CallNamedArgument`
+* `CallNamedArgumentName`
+* `CallNamedArgumentValue`
+* `BooleanLiteral`
+* `ByteLiteral`
+* `ByteStringLiteral`
+* `CharacterLiteral`
+* `ListLiteral`
+* `MapLiteral`
+* `NullLiteral`
+* `NumberLiteral`
+* `RegexpLiteral`
+* `SetLiteral`
+* `StringLiteral`
+* `TupleLiteral`
+* `TypeLiteral`
+* `OtherLiteral`
+* `MapEntry`
+* `MapKey`
+* `MapValue`
+* `PrimitiveType`
+* `AssignmentVariable`
+* `AssignmentValue`
+* `AugmentedAssignment`
+* `AugmentedAssignmentOperator`
+* `AugmentedAssignmentVariable`
+* `AugmentedAssignmentValue`
+
+The following roles are proposed for inclusion:
+
+* `Identifier`
+* `Qualified`
+* `Operator`
+* `Binary`
+* `Unary`
+* `Left`
+* `Right`
+* `Bitwise`
+* `Boolean`
+* `Unsigned`
+* `LeftShift`
+* `RightShift`
+* `Or`
+* `Xor`
+* `And`
+* `Equal`
+* `Not`
+* `LessThan`
+* `LessThanOrEqual`
+* `GreaterThan`
+* `GreaterThanOrEqual`
+* `Identical`
+* `Contains`
+* `Increment`
+* `Decrement`
+* `Negative`
+* `Positive`
+* `Dereference`
+* `TakeAddress`
+* `Add`
+* `Substract`
+* `Multiply`
+* `Divide`
+* `Modulo`
+* `Package`
+* `Declaration`
+* `Import`
+* `Path`
+* `Alias`
+* `Function`
```

Comment:
we don't have a Class role?

Reply:
That would be `TypeDeclaration` or `Type+Declaration` with the new roles.